src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/EngineGeneralTest.java

```diff
@@ -2619,6 +2619,45 @@ public void updateWithWhereTest() throws DatabaseEngineException {
         );
     }
 
+    @Test
+    public void updateFrom1ColTest() throws DatabaseEngineException {
+        test5Columns();
+        test1Column();
+        engine.persist("TEST", entry().set("COL1", 1).set("COL5", "teste")
+                                      .build());
+        engine.persist("TEST", entry().set("COL1", 2).set("COL5", "xpto")
+                                      .build());
+        engine.persist("TEST", entry().set("COL1", 3).set("COL5", "xpto")
+                                      .build());
+        engine.persist("TEST", entry().set("COL1", 4).set("COL5", "teste")
+                                      .build());
+        engine.persist("TEST", entry().set("COL1", 5).set("COL5", "pomme de terre")
+                                      .build());
+
+        engine.persist("TEST2", entry().set("COL1", 1).set("COL2", "update1")
+                                      .build());
+        engine.persist("TEST2", entry().set("COL1", 5).set("COL2", "update2")
+                                      .build());
+
+        final Update updateFrom =
+                update(table("TEST"))
+                        .from(table("TEST2"))
+                        .set(eq(column("COL5"), column("TEST2", "COL2")))
+                        .where(eq(column("TEST", "COL1"), column("TEST2", "COL1")));
+
+        engine.executeUpdate(updateFrom);
+
+        // check to see if TEST has changed
+        final Query query = select(column("COL5"))
+                .from(table("TEST"))
+                .where(in(column("COL1"), L(k(1), k(5))));
+
+        final List<Map<String, ResultColumn>> result = engine.query(query);
+
+        assertEquals("update1", result.get(0).get("COL5").toString());
+        assertEquals("update2", result.get(1).get("COL5").toString());
```

Comment:
Maybe we should select all rows, to assert that those which shouldn't have been updated are keeping their initial values.